lib/carrier/messaging/connection.ex

```diff
@@ -96,7 +96,21 @@ defmodule Carrier.Messaging.Connection do
   """
   def publish(conn, message, kw_args) when is_map(message) do
     topic = Keyword.fetch!(kw_args, :routed_by)
-    :emqttc.publish(conn, topic, Poison.encode!(message))
+
+    encoded = Poison.encode!(message)
+    case Keyword.fetch(kw_args, :threshold) do
+      {:ok, threshold} ->
+        size = byte_size(encoded)
+        if size > threshold do
+          Logger.warn("Message potentially too long (#{size} bytes)")
```

Comment:
It'll vary by chat provider and we don't really have a way to handle that cleanly yet.